### PR TITLE
Adds integration with document signing service from dokobit

### PR DESCRIPTION
**Pull request by gudjong**
# Wraps Dokobit's document signing API

## What

A service for signing PDF documents using Dokobit's API. The service can be extended later to include other types of documents and/or other capabilities supported by Dokobit's API.

## Why

This service will first be used to sign verdicts in detention request cases.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/dokobit-signing/src/lib/signing.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    name: testDocumentName,
+    digest: createHash('sha256')
+      .update(testSignedDocumentContent)
+      .digest('hex'),
+    content: testSignedDocumentContent,
+  },
+}
+const postMock = jest.fn(function(
+  path: string,
+  body?: Body, // eslint-disable-line @typescript-eslint/no-unused-vars
```
Comment (oddsson) [inline libs/dokobit-signing/src/lib/signing.service.spec.ts:42]: Why include these parameters if we don't use them?

File: libs/dokobit-signing/src/lib/signing.service.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    )
+
+    const base64 = Base64.encode(documentContent)
+    const digest = createHash('sha256')
+      .update(base64)
+      .digest('hex')
+
+    const body = new FormData()
+    body.append('phone', mobileNumber)
+    body.append('message', `${message} `)
```
Comment (oddsson) [inline libs/dokobit-signing/src/lib/signing.service.ts:83]: Please consider replacing 

```
 body.append('message', `${message} `)
```

with
 
```
 body.append('message', message)
```

for code readability

File: libs/dokobit-signing/src/lib/signing.service.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+  async getDocument(
+    documentName: string,
+    documentToken: string,
+  ): Promise<string> {
+    let resStatus: DokobitStatusResponse
+
+    // Try to retrieve the signed document
+    // Need to try longer than the mobile signature timeout, but not too long
+    for (let i = 1; i < 120; i++) {
```
Comment (oddsson) [inline libs/dokobit-signing/src/lib/signing.service.ts:116]: This seems like a code smell. Why do we need this  if we are awaiting `this.post`

File: libs/nova-sms/src/lib/sms.service.spec.ts [hunk 4]
```diff
@@ -12,8 +12,8 @@ const testSendSms = 'SendSms'
 const testCode = '0'
 const postMock = jest.fn(function(
   path: string,
-  body?: Body,
-  init?: RequestInit,
+  body?: Body, // eslint-disable-line @typescript-eslint/no-unused-vars
```
Comment (oddsson) [inline libs/nova-sms/src/lib/sms.service.spec.ts:15]: Same question as above ☝️

**Review by oddsson — CHANGES_REQUESTED** [review]

File: libs/dokobit-signing/src/lib/signing.service.spec.ts (hunk 1 above)
Comment (gudjong) [reply libs/dokobit-signing/src/lib/signing.service.spec.ts:42]: This is a mock of a function and the mock does not grab the call if I exclude the arguments. Later, I match calls to the mocked function.

File: libs/dokobit-signing/src/lib/signing.service.ts (hunk 2 above)
Comment (gudjong) [reply libs/dokobit-signing/src/lib/signing.service.ts:83]: I need the extra space at the end. A four digit security code is added to the end of this message on the mobile phone. Without the space, the display will be kind of ugly.

File: libs/nova-sms/src/lib/sms.service.spec.ts (hunk 4 above)
Comment (gudjong) [reply libs/nova-sms/src/lib/sms.service.spec.ts:15]: Same answer.

Comment (oddsson) [reply libs/nova-sms/src/lib/sms.service.spec.ts:15]: Same 👍

File: libs/dokobit-signing/src/lib/signing.service.ts (hunk 3 above)
Comment (gudjong) [reply libs/dokobit-signing/src/lib/signing.service.ts:116]: I agree, but this is how the Dokobit API works and how the code examples from Dokobit do this.

The API returns pretty much immediately, whether the document has been signed or not. At the same time, the user gets a long time to respond to the signature request on the mobile phone.

The result is that we need to try repeatedly until we get an answer other than ´waiting´. I don't know of a good way to do this differently.

**Review by oddsson — APPROVED** [review]

**Pull request merged**